### PR TITLE
Rwa 2272

### PR DESCRIPTION
**Pull request by burale**
### JIRA link (if applicable) ###

[RWA-2272](https://tools.hmcts.net/jira/browse/RWA-2272)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by rasikasri — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java [hunk 1]
```diff
@@ -226,102 +228,6 @@ void should_return_500_with_application_problem_response_when_assign_call_fails(
             ));
     }
 
-    @ParameterizedTest
-    @CsvSource(value = {
-        "IA, Asylum",
-        "WA, WaCaseType"
-    })
-    void should_return_500_with_application_problem_response_when_assign_call_fails_due_to_missing_assignee_id(
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskAssignByIdControllerTest.java:234]: This is specifically testing 500 edge case when assignee id is not provided during Assign action. Is this removed for another reason?

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsControllerTest.java [hunk 2]
```diff
@@ -209,41 +195,6 @@ void should_succeed_and_return_a_204_no_content_when_assigning_task_for_null_ass
         assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
     }
 
-    @Test
-    void should_fail_and_return_a_204_no_content_when_assigning_task_for_null_assignee_for_non_granular() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsControllerTest.java:213]: Should this be removed? removing feature flag is unrelated to test scenario.

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskTest.java [hunk 3]
```diff
@@ -331,41 +271,6 @@ void completeTask_should_throw_role_assignment_verification_exception_when_has_a
         verify(camundaService, times(0)).completeTask(any(), anyBoolean());
     }
 
-    @Test
-    void completeTask_should_throw_task_state_incorrect_exception_when_task_has_no_assignee_gp_flag_is_off() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskTest.java:335]: regardless of gp flag, this edge case test should behave the same way. i.e. throw task state incorrect exception. Should this be removed?

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskWithPrivilegeAndCompletionOptionsTest.java [hunk 4]
```diff
@@ -335,40 +276,6 @@ void should_throw_role_assignment_verification_exception_when_has_access_returns
                 verify(camundaService, times(0)).completeTask(any(), anyBoolean());
             }
 
-            @Test
-            void should_throw_role_assignment_verification_exception_when_has_access_returns_false_gp_flag_off() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskWithPrivilegeAndCompletionOptionsTest.java:339]: same as previous comment. we should still be able to test role_assignment_verification exception regardless of gp flag

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskWithPrivilegeAndCompletionOptionsTest.java [hunk 5]
```diff
@@ -389,49 +296,6 @@ void should_throw_task_state_incorrect_exception_when_task_has_no_assignee_gp_fl
...
-                    .hasNoCause()
-                    .hasMessage(
-                        String.format("Could not complete task with id: %s as task was not previously assigned", taskId)
-                    );
-
-                verify(camundaService, times(0)).completeTask(any(), anyBoolean());
-            }
-
-            @Test
-            void should_throw_task_state_incorrect_exception_when_task_has_no_assignee_gp_flag_off() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskWithPrivilegeAndCompletionOptionsTest.java:414]: same as above?

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskWithPrivilegeAndCompletionOptionsTest.java [hunk 6]
```diff
@@ -456,42 +320,6 @@ void should_throw_exception_when_task_resource_not_found_gp_flag_on() {
...
-                    .isInstanceOf(TaskNotFoundException.class)
-                    .hasNoCause()
-                    .hasMessage("Task Not Found Error: The task could not be found.");
-
-                verify(camundaService, times(0)).completeTask(any(), anyBoolean());
-                verify(cftTaskDatabaseService, times(0)).saveTask(taskResource);
-            }
-
-            @Test
-            void should_throw_exception_when_task_resource_not_found_gp_flag_off() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/taskmanagementservicetests/CompleteTaskWithPrivilegeAndCompletionOptionsTest.java:480]: same as previous?

**Review by rasikasri — APPROVED** [review]
